static/js/components/HomePage.jsx.template

```diff
@@ -46,59 +46,15 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-{%- set ns = namespace(xlg=[], lg=[], md=[], sm=[], xs=[]) -%}
-{%- for widgetName, widgetProps in app.homepage_widgets.items() %}
-  {%- if widgetProps.show %}
-    {%- set ns.xlg = [{
-      "i": widgetName,
-      "x": widgetProps.layouts.xlg[0],
-      "y": widgetProps.layouts.xlg[1],
-      "w": widgetProps.layouts.xlg[2],
-      "h": widgetProps.layouts.xlg[3],
-      "isResizable": widgetProps.resizeable}] + ns.xlg
-    -%}
-    {%- set ns.lg = [{
-      "i": widgetName,
-      "x": widgetProps.layouts.lg[0],
-      "y": widgetProps.layouts.lg[1],
-      "w": widgetProps.layouts.lg[2],
-      "h": widgetProps.layouts.lg[3],
-      "isResizable": widgetProps.resizeable}] + ns.lg
-    -%}
-    {%- set ns.md = [{
-      "i": widgetName,
-      "x": widgetProps.layouts.md[0],
-      "y": widgetProps.layouts.md[1],
-      "w": widgetProps.layouts.md[2],
-      "h": widgetProps.layouts.md[3],
-      "isResizable": widgetProps.resizeable }] + ns.md
-    -%}
-    {%- set ns.sm = [{
-      "i": widgetName,
-      "x": widgetProps.layouts.sm[0],
-      "y": widgetProps.layouts.sm[1],
-      "w": widgetProps.layouts.sm[2],
-      "h": widgetProps.layouts.sm[3],
-      "isResizable": widgetProps.resizeable }] + ns.sm
-    -%}
-    {%- set ns.xs = [{
-      "i": widgetName,
-      "x": widgetProps.layouts.xs[0],
-      "y": widgetProps.layouts.xs[1],
-      "w": widgetProps.layouts.xs[2],
-      "h": widgetProps.layouts.xs[3],
-      "isResizable": widgetProps.resizeable }] + ns.xs
-    -%}
-  {% endif -%}
-{% endfor %}
-
 const defaultLayouts = {
-  xlg: {{ ns.xlg | replace("True", "true") | replace("False", "false") }},
-  lg: {{ ns.lg | replace("True", "true") | replace("False", "false") }},
-  md: {{ ns.md | replace("True", "true") | replace("False", "false") }},
-  sm: {{ ns.sm | replace("True", "true") | replace("False", "false") }},
-  xs: {{ ns.xs | replace("True", "true") | replace("False", "false") }},
-};
+  {%- for breakpoint in app.homepage_grid.breakpoints.keys() %}
+  {{ breakpoint }}: [
+    {%- for widgetName, widgetProps in app.homepage_widgets.items() %}
+    { i: "{{ widgetName }}", x: {{ widgetProps.layouts[breakpoint][0] }}, y: {{ widgetProps.layouts[breakpoint][1] }}, w: {{ widgetProps.layouts[breakpoint][2] }}, h: {{ widgetProps.layouts[breakpoint][3] }}, isResizable: {{ widgetProps.resizeable | lower }} },
```

Comment:
Some newlines here may help with readability, but not strictly necessary.

Reply:
Adding newlines here would transfer to the generated JSX like this:
![Screenshot 2020-09-03 215039](https://user-images.githubusercontent.com/17696889/92200748-85917f00-ee2f-11ea-947c-789422b2e724.png)

I thought it made the generated file unnecessarily long (adds like 125 lines extra). Do we want to prioritize readability on the template or the generated JSX?

Reply:
Not a big deal; you can leave as is.